src/overlays/actors/ovl_En_Karebaba/z_en_karebaba.c

```diff
@@ -19,57 +37,480 @@ const ActorInit En_Karebaba_InitVars = {
+        EnKarebaba_SetupAwaken(this);
+    }
+}
+
+void EnKarebaba_Awaken(EnKarebaba* this, GlobalContext* globalCtx) {
+    SkelAnime_FrameUpdateMatrix(&this->skelAnime);
+    Math_ApproxF(&this->actor.scale.x, 0.01f, 0.0005f);
+    this->actor.scale.y = this->actor.scale.z = this->actor.scale.x;
+    if (Math_ApproxF(&this->actor.posRot.pos.y, this->actor.initPosRot.pos.y + 60.0f, 5.0f)) {
+        EnKarebaba_SetupUpright(this);
+    }
+    this->actor.shape.rot.y += 0x1999;
+    EffectSsHahen_SpawnBurst(globalCtx, &this->actor.initPosRot.pos, 3.0f, 0, 12, 5, 1, -1, 10, 0);
+}
+
+void EnKarebaba_Upright(EnKarebaba* this, GlobalContext* globalCtx) {
+    Player* player = PLAYER;
+
+    SkelAnime_FrameUpdateMatrix(&this->skelAnime);
+
+    if (this->actor.params != 0) {
+        this->actor.params--;
+    }
+
+    if (func_800A56C8(&this->skelAnime, 0.0f) || func_800A56C8(&this->skelAnime, 12.0f)) {
+        Audio_PlayActorSound2(&this->actor, NA_SE_EN_DEKU_JR_MOUTH);
+    }
+
+    // Check if Link attacks the Deku Baba
+    if (this->bodyCollider.base.acFlags & 2) {
+        EnKarebaba_SetupDying(this);
+        func_80032C7C(globalCtx, &this->actor);
+    } else if (Math_Vec3f_DistXZ(&this->actor.initPosRot.pos, &player->actor.posRot.pos) > 240.0f) {
+        EnKarebaba_SetupRetract(this);
+    } else if (this->actor.params == 0) {
+        EnKarebaba_SetupSpin(this);
+    }
+}
+
+void EnKarebaba_Spin(EnKarebaba* this, GlobalContext* globalCtx) {
+    s32 value;
+    f32 thing;
+
+    if (this->actor.params != 0) {
+        this->actor.params--;
+    }
+
+    SkelAnime_FrameUpdateMatrix(&this->skelAnime);
+
+    if (func_800A56C8(&this->skelAnime, 0.0f) || func_800A56C8(&this->skelAnime, 12.0f)) {
+        if (1) { } // Here for matching purposes only.
+
+        Audio_PlayActorSound2(&this->actor, NA_SE_EN_DEKU_JR_MOUTH);
+    }
+
+    value = 20 - this->actor.params;
+    value = 20 - ABS(value);
+
+    if (value > 10) {
+        value = 10;
+    }
+
+    this->headCollider.dim.radius = sHeadColliderInit.dim.radius + (value * 2);
+    this->actor.shape.rot.x = 0xC000 - (value * 0x100);
+    this->actor.shape.rot.y += value * 0x2C0;
+    this->actor.posRot.pos.y = (Math_Sins(this->actor.shape.rot.x) * -60.0f) + this->actor.initPosRot.pos.y;
+
+    thing = Math_Coss(this->actor.shape.rot.x) * 60.0f;
+
+    this->actor.posRot.pos.x = (Math_Sins(this->actor.shape.rot.y) * thing) + this->actor.initPosRot.pos.x;
+    this->actor.posRot.pos.z = (Math_Coss(this->actor.shape.rot.y) * thing) + this->actor.initPosRot.pos.z;
+
+    if (this->bodyCollider.base.acFlags & 2) {
+        EnKarebaba_SetupDying(this);
+        func_80032C7C(globalCtx, &this->actor);
+    } else if (this->actor.params == 0) {
+        EnKarebaba_SetupUpright(this);
+    }
+}
+
+void EnKarebaba_Dying(EnKarebaba* this, GlobalContext* globalCtx) {
+    static Vec3f sVecZero1 = { 0.0f, 0.0f, 0.0f };
+
```

Comment:
the `s` prefix for data is only for data that lives outside of functions, id suggest just calling this `zeroVec`, the name we use pretty commonly for this